db/change_cache.go

```diff
@@ -468,13 +469,24 @@ func (c *changeCache) DocChangedSynchronous(event sgbucket.FeedEvent) {
 	}
 
 	if syncData.Sequence <= c.getInitialSequence() {
-		return // Tap is sending us an old value from before I started up; ignore it
+		return // DCP is sending us an old value from before I started up; ignore it
 	}
 
-	// Record a histogram of the Tap feed's lag:
-	tapLag := time.Since(syncData.TimeSaved) - time.Since(entryTime)
-	lagMs := int(tapLag/(100*time.Millisecond)) * 100
-	changeCacheExpvars.Add(fmt.Sprintf("lag-tap-%04dms", lagMs), 1)
+	// Measure feed latency from timeSaved or the time we started working the feed, whichever is later
+	var feedLatency time.Duration
+	if !syncData.TimeSaved.IsZero() {
```

Comment:
Is there an expected case where we'd see this come in as zero? If so, do we want to explicitly exclude those from `StatKeyDcpReceivedCount`, or would we want to track them regardless?

Reply:
I think this is primarily defensive and shouldn't happen, so I thought it was probably preferable to increment the count if there were a case this happens - either the calculated rate or the count stat would end up being off, and I thought the reported count was probably the better one to preserve.